### DISCUSSION TITLE
set device name

### DISCUSSION THREAD

**Discussion by rozrabiak** [dawidchyrzynski/arduino-home-assistant#132] [General]
Hi, is it possible to set mqqt name?
Now in mosqitto I see device name as: ew client connected from 192.168.1.11:60097 as 0010fa6e384a
And I wan't my own name of connected device for e.g. "MyOwnDevice" instead of: 0010fa6e384a
0010fa6e384a is automatically random named?

When I use PubSubClient library I can set my name for device for MQTT.

**Comment by rrozema** [comment]
Did you try calling HADevice's .setName("your device name") in your setup function, somewhere before the call to HAMqtt's .begin( .... ) ?
